proxy/InkAPI.cc

```diff
@@ -9193,3 +9193,71 @@ TSHttpTxnIdGet(TSHttpTxn txnp)
 
   return (uint64_t)sm->sm_id;
 }
+
+// Return information about the protocols used by the client
+TSReturnCode
+TSHttpTxnClientProtocolStackGet(TSHttpTxn txnp, int n, char const **result, int *actual)
+{
+  sdk_assert(sdk_sanity_check_txn(txnp) == TS_SUCCESS);
+  sdk_assert(result != NULL);
+  HttpSM *sm = (HttpSM *)txnp;
+  int count  = 0;
+  if (sm) {
+    count = sm->populate_client_protocol(result, n);
+  }
+  if (actual) {
+    *actual = count;
```

Comment:
Is ``result`` allowed to be ``NULL``? We should ``sdk_assert`` that ``result`` and ``actual`` are both either present or absent together.